auth-providers/oauth2-provider/src/main/java/tech/ydb/auth/OAuth2TokenSource.java

```diff
@@ -0,0 +1,199 @@
+
+    protected OAuth2TokenSource(String type, int expireInSeconds) {
+        this.type = type;
+        this.expireInSeconds = expireInSeconds;
+    }
+
+    public String getType() {
+        return this.type;
+    }
+
+    public int getExpireInSeconds() {
+        return this.expireInSeconds;
+    }
+
+    public abstract String getToken();
+
+    public static OAuth2TokenSource fromValue(String token) {
+        return fromValue(token, JWT_TOKEN);
+    }
+
+    public static OAuth2TokenSource fromValue(String token, String tokenType) {
+        return new OAuth2TokenSource(tokenType, 100 * 365 * 24 * 60 * 60) { // Expire in 100 year ~ never expired
+            @Override
+            public String getToken() {
+                return token;
+            }
+        };
+    }
+
+    public static OAuth2TokenSource fromFile(File tokenFile) {
+        return fromFile(tokenFile, JWT_TOKEN);
+    }
+
+    public static OAuth2TokenSource fromFile(File tokenFile, String tokenType) {
+        return new OAuth2TokenSource(tokenType, 24 * 60 * 60) { // Expire in 1 day
+            @Override
+            public String getToken() {
+                try (FileReader reader = new FileReader(tokenFile)) {
+                    return CharStreams.toString(reader).trim();
+                } catch (IOException e) {
+                    throw new RuntimeException("Unable to read token from " + tokenFile, e);
+                }
+            }
+        };
+    }
+
+    public static JWTTokenBuilder fromPrivateKeyPemFile(File privateKeyPemFile) {
+        try {
+            StringBuilder base64 = new StringBuilder();
+            try (BufferedReader br = new BufferedReader(new FileReader(privateKeyPemFile))) {
+                for (String line = br.readLine(); line != null; line = br.readLine()) {
+                    if (line.startsWith("-----")) {
+                        continue;
+                    }
+                    base64.append(line);
+                }
+            }
+            byte[] bytes = Base64.getDecoder().decode(base64.toString());
+            KeyFactory kf = KeyFactory.getInstance("RSA");
+            PrivateKey key = kf.generatePrivate(new PKCS8EncodedKeySpec(bytes));
+            return new JWTTokenBuilder(key);
+        } catch (IOException | NoSuchAlgorithmException | InvalidKeySpecException e) {
+            throw new RuntimeException("Unable to read key from " + privateKeyPemFile, e);
+        }
+    }
+
+    public static JWTTokenBuilder fromKey(Key key) {
+        return new JWTTokenBuilder(key);
+    }
+
+    public static class JWTTokenBuilder {
+        private final Key signingKey;
+        private final SignatureAlgorithm alg;
+
+        private Clock clock = Clock.systemUTC();
+        private int ttlSeconds = 60 * 60; // 1 hour by default
+
+        private String issuer = null;
+        private String subject = null;
+        private String audience = null;
+        private String id = null;
+
```

Comment:
Also there is an important header claim: kid: https://github.com/ydb-platform/ydb-python-sdk/blob/main/ydb/oauth2_token_exchange/token_source.py#L96